src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VQSRTrainingSetManager.java

```diff
@@ -0,0 +1,144 @@
+         */
+        public void annotateDatum(FeatureContext tracker, VariantContext evalVC, VariantDatum datum, boolean trust_all_polymorphic) {
+            for( final VariantContext trainVC : tracker.getValues(this.getFeatureInput()) ) {
+                if( isValidVariant( evalVC, trainVC, trust_all_polymorphic ) ) {
+                    datum.isKnown = datum.isKnown || this.isKnown;
+                    datum.atTruthSite = datum.atTruthSite || this.isTruth;
+                    datum.atTrainingSite = datum.atTrainingSite || this.isTraining;
+                    datum.prior = Math.max( datum.prior, this.prior );
+                    datum.consensusCount += ( this.isConsensus ? 1 : 0 );
+                }
+                if( trainVC != null ) {
+                    datum.atAntiTrainingSite = datum.atAntiTrainingSite || this.isAntiTraining;
+                }
+            }
+        }
+
+        private static boolean isValidVariant( final VariantContext evalVC, final VariantContext trainVC, final boolean trustAllPolymorphic) {
+            return trainVC != null && trainVC.isNotFiltered() && trainVC.isVariant() && checkVariationClass( evalVC, trainVC ) &&
+                    (trustAllPolymorphic || !trainVC.hasGenotypes() || trainVC.isPolymorphicInSamples());
+        }
+
+        private static boolean checkVariationClass( final VariantContext evalVC, final VariantContext trainVC ) {
+            switch( trainVC.getType() ) {
+                case SNP:
+                case MNP:
+                    return evalVC.isSNP() || evalVC.isMNP();
+                case INDEL:
+                case MIXED:
+                case SYMBOLIC:
+                    return evalVC.isStructuralIndel() || evalVC.isIndel() || evalVC.isMixed() || evalVC.isSymbolic();
+                default:
+                    return false;
+            }
+        }
+
+        public FeatureInput<VariantContext> getFeatureInput(){
+            return fi;
+        }
+    }
+
+    /**
+     * Adds data sets to this manager.
+     */
+    public void addDataSets(final Collection<FeatureInput<VariantContext>> dataSetsArgs) {
+        dataSetsArgs.forEach(this::addDataSet);
+    }
+
+    /**
+     * Adds a data set to this manager.
+     */
+    public void addDataSet(final FeatureInput<VariantContext> dataset) {
+        dataSets.add(new DataSet(dataset));
+    }
+
+    /**
+     * Returns whether the manager has a data set that is marked as 'training'.
+     */
+    public boolean hasTrainingSet() {
+        return dataSets.stream().anyMatch(ts -> ts.isTraining);
+    }
+
+    /**
+     * Returns whether the manager has a data set that is marked as 'truth'.
+     */
+    public boolean hasTruthSet() {
+        return dataSets.stream().anyMatch(ts -> ts.isTruth);
+    }
+
+    public void annotateDatum(final FeatureContext tracker, final VariantContext evalVC, final VariantDatum datum, final boolean trust_all_polymorphic) {
```

Comment:
Javadoc here explaining what this is doing, since it's unusual to pass a parameter around a to be modified like this.